src/components/DualTimeline.tsx

```diff
@@ -0,0 +1,376 @@
+  const trackTitle = 'mmm'
+
+  return (
+    <Container>
+      <LineStartsEnds isSingle starts headings={[trackTitle]} />
+      {Object.entries(years).map(([year, values]) => (
+        <>
+          <Year value={year} headings={[trackTitle]} isSingle />
+          {values.map(({ title, entries }) => (
+            <>
+              {entries && (
+                <>
+                  <Entries>
+                    <Entry>
+                      <Track forHeading={trackTitle} />
+                      <Heading isSingle>{title}</Heading>
+                    </Entry>
+                  </Entries>
+                  <LineStartsEnds isSingle height="2rem" headings={[trackTitle]} />
+                </>
+              )}
+              {entries?.map((entry, index) => (
+                <Entry isSingle key={index}>
+                  <Track forHeading={[trackTitle]} entry={entry} />
+                  <Data>
+                    <Text isMajor={entry?.isMajor}>{entry.text}</Text>
+                    <When>{entry.when}</When>
+                  </Data>
+                </Entry>
+              ))}
+            </>
+          ))}
+        </>
+      ))}
+      <LineStartsEnds isSingle headings={[trackTitle]} />
+      <LineStartsEnds isSingle ends headings={[trackTitle]} />
+    </Container>
+  )
+}
+
+const Container = styled.div`
+  width: 100%;
+  margin-bottom: 226px;
+`
+
+const Text = styled.div`
+  ${({ isMajor }) =>
+    isMajor &&
+    `
+    background: linear-gradient(180deg, #00C2FF 40%, #2400FF 100%);
+    -webkit-background-clip: text;
+    -webkit-text-fill-color: transparent;
+    -moz-background-clip: text;
+    -moz-text-fill-color: transparent;
+  `}
+`
+const When = styled.div`
+  color: #555555;
+`
+
+const Dot = styled.div`
+  position: relative;
+  left: -9px;
+  top: calc((18px + 4px) / -2);
+
+  ::after {
+    width: 18px;
+    height: 18px;
+    border-radius: 100%;
+    background: ${({ isMajor }) => (isMajor ? 'linear-gradient(180deg, #00C2FF 0%, #2400FF 100%);' : 'white')};
+    border: 4px solid black;
+    position: absolute;
+    content: '';
+    display: block;
+  }
+`
+
+const YearLine = styled.div`
+  height: 1px;
+  background-color: grey;
+  width: 100%;
+  position: absolute;
+  z-index: -1;
+`
+
+const Data = styled.div`
+  display: flex;
+  line-height: 22px;
+  flex-direction: column;
+  text-align: ${({ right }) => (right ? 'right' : 'left')};
+  padding-bottom: 54px;
+
+  ${({ isSingle }) =>
+    isSingle &&
+    `
+    position: relative;
+    ${({ right }) => (right ? 'right: -3rem;' : 'left: -2rem')}
+  `}
+`
+
+const Piece = styled.div`
+  display: flex;
+  align-items: center;
+  width: 8px;
+  height: 100%;
+  background-color: #555555;
+  padding-bottom: 54px;
+`
+
+const YearData = styled.div`
+  width: 100%;
+  padding-bottom: 3rem;
+  font-weight: var(--fontWeight-semiBold);
+  font-size: var(--fontSize-28);
+`
+
+let Year = ({ value, headings, isSingle }) =>
```

Comment:
We're mixing components and styles. Our internal rule so far, was to keep all stylings at the end of the file.